intbot/core/integrations/github.py

```diff
@@ -55,127 +77,175 @@ def as_discord_message(self):
         return f"[{self.title}]({self.url})"
 
 
-@dataclasses.dataclass
-class GithubDraftIssue:
+class GithubDraftIssue(BaseModel):
     id: str
     title: str
 
     def as_discord_message(self):
         return self.title
 
 
-def fetch_github_item_details(item_id):
-    headers = {
-        "Authorization": f"Bearer {settings.GITHUB_API_TOKEN}",
-        "Content-Type": "application/json",
-    }
-    payload = {"query": query, "variables": {"itemId": item_id}}
-    response = httpx.post(GITHUB_API_URL, json=payload, headers=headers)
-    if response.status_code == 200:
-        return response.json()["data"]["node"]["content"]
-    else:
-        raise Exception(f"GitHub API error: {response.status_code} - {response.text}")
+class GithubWebhook:
+    """
+    Base class for all the other specific types of webhooks.
+    """
 
+    def __init__(self, action: str, headers: dict, content: dict, extra: dict):
+        self.action = action
+        self.headers = headers
+        self.content = content
+        self.extra = extra
+        self._project = None
+        self._repository = None
+
+    @classmethod
+    def from_webhook(cls, wh: Webhook):
+        assert wh.extra, "Extra should be set already at this point"
```

Comment:
Why is `assert` outside of test file? Looking at a random articles on the internet, it's an antipatterns and the code would break. If that's the intention, let's use proper error messages, so if it breaks, we know why

Reply:
I'm going to keep it, because we never supposed to create GithubWebhook from Webhook without extra (for business logic reasons). We should probably document it better why, but the only other option here is to raise a different exception that AssertionError.

Reply:
Note to self: Let's add architecture.md to explain how the github flow works.

Reply:
Added #12